### PR TITLE
[#359] further tests to validate reference to existing config and spl…

### PR DESCRIPTION
**Pull request by gtully**
…it console realm

### CODE REVIEW

**Review by gaohoward — APPROVED** [review]

File: controllers/activemqartemis_controller.go [hunk 1]
```diff
@@ -461,6 +461,19 @@ func UpdateCRStatus(cr *brokerv1beta1.ActiveMQArtemis, client rtclient.Client, n
...
+	if len(cr.Status.ExternalConfigs) >= 0 {
+		for _, cfg := range cr.Status.ExternalConfigs {
+			for _, curCfg := range current.Status.ExternalConfigs {
+				if curCfg.Name == cfg.Name && curCfg.ResourceVersion != cfg.ResourceVersion {
+					return resources.UpdateStatus(client, cr)
+				}
+			}
+		}
+	}
+
```
Comment (brusdev) [inline controllers/activemqartemis_controller.go:476]: @gtully could you move this change in a separate commit?

Comment (gtully) [reply controllers/activemqartemis_controller.go:476]: yep, either that or verify the external config status in one of those tests. I omitted doing that. let me revisit in the next day or so. thanks for the feedback.

**Comment by gtully** [comment]
I am getting an error in make test-mk that I need to track down.

File: controllers/activemqartemis_controller.go (hunk 1 above)
Comment (gtully) [reply controllers/activemqartemis_controller.go:433]: I have added a test to validate the need for that fix. thanks!

**Comment by gtully** [comment]
Dom found the problem with the tests, the mad naming of crs fell down when replaced chars left a nothing between dashes... and that was an invalid field in the jolokia policy. Time to name the test crs after their line numbers!

**Comment by brusdev** [comment]
@gtully I created the #477 PR to improve the suite test output and change the resources names.

**Review by brusdev — APPROVED** [review]
